lib/Serialization/ModuleFile.cpp

```diff
@@ -291,7 +291,7 @@ void ModuleFile::lookupValue(DeclName name,
         Expected<Decl *> declOrError = getDeclChecked(item.second);
         if (!declOrError) {
           if (!getContext().LangOpts.EnableDeserializationRecovery)
-            fatal(declOrError.takeError());
+            return diagnoseAndConsumeFatal(declOrError.takeError());
```

Comment:
There shouldn't be any recovery when `LangOpts.EnableDeserializationRecovery` is disabled. That flag is present to help in debugging failures that are wrongfully recovered from, so please make sure to keep the hard errors in such a case. This flag should always be enabled outside of tests and manual deserialization failure investigation.

Reply:
I assume you meant to write it's always _disabled_ outside of tests?

Reply:
Assuming that's what you meant, I think I addressed this comment now :-)

Reply:
I meant `EnableDeserializationRecovery` is enabled by default. We only disable recovery in testing and during investigations.